src/node/internal/internal_dns.ts

```diff
@@ -0,0 +1,264 @@
+
+export function getServers(): string[] {
+  throw new Error('Not implemented');
+}
+
+export function lookup(): void {
+  throw new Error('Not implemented');
+}
+
+export function lookupService(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve4(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve6(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveAny(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolveCname(name: string): Promise<string[]> {
+  validateString(name, 'name');
+
+  // Validation errors needs to be sync.
+  // Return a promise rather than using async qualifier.
+  return sendDnsRequest(name, 'CNAME').then((json) => {
+    if (!('Answer' in json)) {
+      // DNS request should contain an "Answer" attribute, but it didn't.
+      throw new DnsError(name, errorCodes.NODATA, 'queryCname');
+    }
+
+    return json.Answer.map((a) => {
+      // Cloudflare DNS returns "nodejs.org." whereas
+      // Node.js returns "nodejs.org" as a CNAME data.
+      return a.data.slice(0, -1);
+    });
+  });
+}
+
+export function resolveCaa(
+  name: string
+): Promise<
+  { critical: number; issue?: string; iodef?: string; issuewild?: string }[]
+> {
+  validateString(name, 'name');
+
+  // Validation errors needs to be sync.
+  // Return a promise rather than using async qualifier.
+  return sendDnsRequest(name, 'CAA').then((json) => {
+    if (!('Answer' in json)) {
+      // DNS request should contain an "Answer" attribute, but it didn't.
+      throw new DnsError(name, errorCodes.NOTFOUND, 'queryCaa');
+    }
+
+    // CAA API returns "hex", so we need to convert it to UTF-8
+    return json.Answer.map((a) => {
+      const obj = { critical: 0 };
+      const record = dnsUtil.parseCaaRecord(a.data);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error
+      obj[record.field] = record.value;
+      obj.critical = record.critical;
+      return obj;
+    });
+  });
+}
+
+export function resolveMx(
+  name: string
+): Promise<{ exchange: string; priority: number }[]> {
+  validateString(name, 'name');
+
+  // Validation errors needs to be sync.
+  // Return a promise rather than using async qualifier.
+  return sendDnsRequest(name, 'MX').then((json) => {
+    if (!('Answer' in json)) {
+      // DNS request should contain an "Answer" attribute, but it didn't.
+      throw new DnsError(name, errorCodes.NOTFOUND, 'queryMx');
+    }
+
+    // Cloudflare API returns "data": "10 smtp.google.com." hence
+    // we need to parse it.
+    return json.Answer.map((a) => {
+      const [priority, value]: string[] = a.data.split(' ');
+      if (priority == null || value == null) {
+        throw new DnsError(name, errorCodes.BADRESP, 'queryMx');
+      }
+      return {
+        exchange: value.slice(0, -1),
```

Comment:
I assume this is also because the value may end in a "."? If so, a comment here would be helpful along with a double check to make sure the value ends with the "." before slicing it.